[Doc]Contribution Guideline updates (#3114)
Azure/communication-ui-library: docs/contributing-guide/6. submitting-a-pr.md

### Prerequisites to publishing a Pull Request

Once you have gotten setup with the repo, made and tested your changes, there are some checks before submitting your PR:

1. Update dependencies lock file
2. Update package API files
3. Generate required change files

#### 1. Update dependencies lock file for all flavors
If you are adding/updating dependencies in any package.json file, you will need to update dependencies lock files for both beta and stable flavors before submitting changes, run these commands to ensure they are correctly generated:

For beta:
```bash
rush update
```

For stable:
```bash
rush update:stable
```

> These commands will update your dependencies locally and generate pnpm-lock.yaml. You might wanna install the right dependencies for your current flavor (by default, run `rush switch-flavor:beta` again for switching back to beta flavor).

Reviewer: ```suggestion
These commands will update your dependencies locally and update the pnpm-lock.yaml. You may wish to install the correct dependencies for your current flavor (by default, run `rush switch-flavor:beta` again for switching back to beta flavor).
```

Reviewer 2: +1

#### 2. Update package API files

To ensure we do not break any existing public APIs, we use [api-extractor](../infrastructure/api-extractor.md) across all our npm packages. This generates a file detailing all public exports of the package. You can find this file under `package-root/review/beta/package-name.api.md`. We do this to prevent any accidental breaking changes to the packages we export, and to ensure we do not accidentally publish any internal helper classes/functions.

When a package is built the api-extractor is automatically run and will update the corresponding `api.md` file.

> - If you are updating apis under stable version, api update for both beta&stable would be necessary, a quick command to do so is `rush build:all-flavors`, which will build under all flavors and generate api files for both. Or you could also switch to stable flavor by running `rush switch-flavor:stable` and do the step below to generate stable api.md (don't forget to run `rush switch-flavor:beta` when finished if necessary).

Reviewer: ```suggestion
- If you are updating APIs for the stable (released) version, API updates for both beta&stable is necessary. A quick command to do so is `rush build:all-flavors`, which will build all flavors and generate API files for both. You could also switch to the stable flavor by running `rush switch-flavor:stable` and perform the step below to generate stable api.md (don't forget to run `rush switch-flavor:beta` when finished if necessary).
```

Author: Is `stable` == `release` version? If I remember correct, we have `beta release`(https://github.com/Azure/communication-ui-library/releases) version over night

> To ensure these API files have been updated, run `rush build -t "package-name"` for each package you have made changes to and submit any changed `api.md` files along with your PR.

Reviewer 3: Might be simpler to just always build the npm package as this will build all deps and generate all api files:

```suggestion
To ensure these API files have been updated, run `rush build -t @azure/communication-react` to have all packlets update their api files and submit any changed `api.md` files along with your PR.
```
?

Author: Yeah, I think having one command might be simpler for contributors

> To ensure these API files have been updated, run `rush build -t @azure/communication-react` to have all packages update their api files and submit any changed `api.md` files along with your PR.

Reviewer 4: ```suggestion
To ensure these API files have been updated, run `rush build -t @azure/communication-react` to have all packages update their api files. Then submit any changed `api.md` files along with your PR.
```

This will start an interactive tool that will:

> 1. First ask you to select whether your change is a Major, Minor, or Patch (for more information on which to choose see [Semantic Versioning](https://semver.org/)).

Reviewer 4: ```suggestion
1. Ask you to select whether your change is a Major, Minor, or Patch (for more information on which to choose see [Semantic Versioning](https://semver.org/)).
```

1. Then ask for a short description of your change - this is what will be included in the changelog when a new version is released, so keep it concise and meaningful. See [Tips for writing meaningful changelog entries](../reviewer-notes/tips-for-writing-changelog-entries).

### Generating new snapshots for your UI changes

If you are making a UI change in your PR, you might invalidate some screen snapshots in our automated visual tests.

> 1. Check the failing Github Actions to see if the failure is due to a UI change.

Reviewer 4: ```suggestion
1. Check the failing Github Actions to see if the failure was caused by a UI change.
```

Author: It could have better consistency here by using `is` because we are keep using `present tense` in this paragraph

   ![screenshot showing downloadable snapshot artifacts in a failed github action](../images/check-snapshots.png)

> 1. Add `update_snapshot` label to the PR.

Reviewer: ```suggestion
1. Add the `update_snapshot` label to the PR.
```

Reviewer 2: +1

1. Wait for Github Actions to generate new snapshots for the PR.

> 1. The validation pipelines will be re-run after new snapshots are generated.

Reviewer 4: ```suggestion
1. The validation pipelines will re-run after new snapshots are generated.
```

Author: I think the use of "be" is necessary in this context to indicate the future action of the validation pipelines being re-run
